profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc

```diff
@@ -12,6 +12,27 @@
  */
 class TMGMTPoetryJobController extends TMGMTJobController {
 
+  /**
+   * When using #cart, delete all jobs and sub jobs (job items).
+   */
+  public function delete($ids, $transaction = NULL) {
+
+    if (!is_array($ids)) {
+      $ids = [$ids];
+    }
+
+    $ids_iterate = $ids;
+    foreach ($ids_iterate as $id_iterate) {
+      $related_jobs = tmgmt_poetry_obtain_related_translation_jobs(array(), 'SUB_' . $id_iterate, TRUE);
+      foreach ($related_jobs as $related_job) {
```

Comment:
Looping over an object is "ugly", there should be a better way to do that I guess ?

Reply:
I will double check... thx @drupol!

Reply:
Yes, it's better to use "->fetchAll()" like in other places this function is used.